packages/@vuepress/core/src/app/prepare/preparePagesRoutes.ts

```diff
@@ -17,10 +19,20 @@ import { Vuepress } from '@vuepress/client/lib/components/Vuepress'
 
 const routeItems = [\
 ${app.pages
-  .map(({ key, path, pathInferred, filePathRelative, title }) => {
+  .map(({ key, path, pathInferred, filePathRelative, title, meta = null }) => {
     const redirects: string[] = []
     const routeItem: RouteItem = [key, path, title, redirects]
 
+    if (meta) {
+      if (JSON.stringify(meta).length <= 1024) {
+        routeItem.push(meta)
+      } else {
+        logger.warn(
+          `[Meta size check]: ${path} meta size > 1kb and will be dropped`
+        )
+      }
+    }
```

Comment:
I think this check is not necessary. Users should bear their own risks about scalability if they determine to inject extra meta into routes. We only need to mention that in the docs.